### PR TITLE
Added rpc api for chain_getCoinbase

### LINKED ISSUE #228: Add a RPC API to query coinbase address
`chain_coinbase`: Returns the client coinbase address.

### PR DESCRIPTION
**Pull request by leekt216**
fixed #228
run
```
$ ./target/release/codechain -c blake_pow --author <YOUR ADDRESS>

$ curl \
    -H 'Content-Type: application/json' \
    -d '{"jsonrpc": "2.0", "method": "chain_getCoinbase", "params": [], "id": null}' \
    localhost:8080
```

### CODE REVIEW

**Comment by leekt216** [comment]
@joojis fixed it to return nill if miner.author is zero

**Comment by leekt216** [comment]
@joojis changed to return null if minor.author is zero

File: rpc/src/v1/impls/chain.rs [hunk 1]
```diff
@@ -144,4 +144,18 @@ where
...
+            Ok(None)
+        } else {
+            Ok(Some(self.miner.author()))
+        }
+        /*
+        Ok() => match self.miner.author().is_zero() {
+            true => Some(self.miner.author()),
+            false => None
+        }
+        */
```
Comment (joojis) [inline rpc/src/v1/impls/chain.rs]: Could you remove this comment?

**Comment by leekt216** [comment]
@joojis fixed it

**Comment by kseo** [comment]
@leekt216 Don't forget to format your code. 

```
cargo +nightly-2018-07-17 fmt
```

**Pull request merged**